modules/sdk-core/src/bitgo/wallet/wallets.ts

```diff
@@ -297,15 +316,59 @@ export class Wallets implements IWallets {
       walletParams.enterprise = enterprise;
     }
 
+    // Validate referenceWalletId for EVM keyring
+    if (!_.isUndefined(referenceWalletId)) {
+      if (!_.isString(referenceWalletId)) {
+        throw new Error('invalid referenceWalletId argument, expecting string');
+      }
+      if (!this.baseCoin.isEVM()) {
+        throw new Error('referenceWalletId is only supported for EVM chains');
+      }
+    }
+
     // EVM TSS wallets must use wallet version 3, 5 and 6
+    // Skip this validation for EVM keyring wallets as they inherit version from reference wallet
     if (
       isTss &&
       this.baseCoin.isEVM() &&
+      !referenceWalletId &&
       !(params.walletVersion === 3 || params.walletVersion === 5 || params.walletVersion === 6)
     ) {
       throw new Error('EVM TSS wallets are only supported for wallet version 3, 5 and 6');
     }
 
+    // Handle EVM keyring wallet creation with referenceWalletId
+    if (isEVMWithReference) {
+      // For EVM keyring wallets, multisigType will be inferred from the reference wallet
+      // No need to explicitly validate TSS requirement here as it will be handled by bgms
+
+      // For EVM keyring wallets, we use the add method directly with referenceWalletId
+      // This bypasses the normal key generation process since keys are shared via keyring
+      const addWalletParams = {
+        label,
+        referenceWalletId,
+      };
+
+      const newWallet = await this.bitgo.post(this.baseCoin.url('/wallet/add')).send(addWalletParams).result();
+
+      // For EVM keyring wallets, we need to get the keychains from the reference wallet
+      const userKeychain = this.baseCoin.keychains().get({ id: newWallet.keys[KeyIndices.USER] });
+      const backupKeychain = this.baseCoin.keychains().get({ id: newWallet.keys[KeyIndices.BACKUP] });
+      const bitgoKeychain = this.baseCoin.keychains().get({ id: newWallet.keys[KeyIndices.BITGO] });
+
+      const [userKey, backupKey, bitgoKey] = await Promise.all([userKeychain, backupKeychain, bitgoKeychain]);
+
+      const result: WalletWithKeychains = {
+        wallet: new Wallet(this.bitgo, this.baseCoin, newWallet),
+        userKeychain: userKey,
+        backupKeychain: backupKey,
+        bitgoKeychain: bitgoKey,
+        responseType: 'WalletWithKeychains',
+      };
+
+      return result;
+    }
```

Comment:
This should also be an EVM lib util i.e something like `createEvmKeyRingWallet`.